src/lib/eventListeners.js

```diff
@@ -31,24 +32,46 @@ export default {
       componentsMap.set(component, components)
     }
 
-    components.add(cb)
+    components.add({ cb, priority })
+    cache.delete(event) // Invalidate the cache when a new callback is added
   },
+
   executeListeners(event, params) {
     const componentsMap = eventsMap.get(event)
-    if (componentsMap) {
-      componentsMap.forEach((component) => {
-        component.forEach((cb) => {
-          cb(params)
-        })
+    if (!componentsMap) {
```

Comment:
we should change this to an `undefined` check instead of a generic falsy check

Reply:
fixed this. but I realized I need to add a check for also the size of componentsMap.